### PR TITLE
Senate.gov website downloading fixes

### PR DESCRIPTION
**Pull request by Grabien**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fix of non-working senateisvp plugin. Also, added support for other kinds of senate.gov website links. Examples of links:

https://www.help.senate.gov/hearings/vaccines-saving-lives-ensuring-confidence-and-protecting-public-health
https://www.appropriations.senate.gov/hearings/watch?hearingid=B8A25434-5056-A066-6020-1F68CB75F0CD
https://www.judiciary.senate.gov/hearings/watch?hearingid=EAA07214-5056-A066-60E5-1B701A8D462E
https://www.banking.senate.gov/hearings/21st-century-communities-public-transportation-infrastructure-investment-and-fast-act-reauthorization
https://www.armed-services.senate.gov/hearings/the-posture-of-the-department-of-the-navy-in-review-of-the-defense-authorization-request-for-fiscal-year-2022-and-the-future-years-defense-program

### CODE REVIEW

File: yt_dlp/extractor/senategov.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        video_type = qs['type'][0]
+        committee = video_type if video_type == 'arch' else qs['comm'][0]
+
+        stream_num, domain = _COMMITTEES[committee]
+
+        formats = []
+        if video_type == 'arch':
+            filename = video_id if '.' in video_id else video_id + '.mp4'
+            formats = [{
+                'url': compat_urlparse.urljoin(domain, 'i/' + filename + '/master.m3u8'),
```
Comment (pukkandan) [inline yt_dlp/extractor/senategov.py:129]: this is wrong and I don't understand what you intended here

Comment (Grabien) [reply yt_dlp/extractor/senategov.py:128]: This is not my code. This is the code of the old "ISVP" extractor from the Youtube-dl code. I have no idea what this code does.

Comment (pukkandan) [reply yt_dlp/extractor/senategov.py:128]: No, u modified this line. The diff is being masked by the rename. See the diff of the original commit you made.
```diff
-                'url': compat_urlparse.urljoin(domain, filename) + '?v=3.1.0&fp=&r=&g=',
+                'url': compat_urlparse.urljoin(domain, 'i/' + filename + '/master.m3u8'),
```

Comment (Grabien) [reply yt_dlp/extractor/senategov.py:128]: Sorry, you are right. This is a bugfix for players like this:
https://www.senate.gov/isvp/?type=arch&comm=intel&filename=intel090613&hc_location=ufi

URLs that had this format:
http://ussenate-f.akamaihd.net/intel090613.mp4

Now have the following format:
http://ussenate-f.akamaihd.net/i/intel090613.mp4/master.m3u8

Comment (pukkandan) [reply yt_dlp/extractor/senategov.py:128]: Did you actually try downloading with this? You cannot simply pass the m3u8 url as a format. It needs to be parsed accordingly

Comment (Grabien) [reply yt_dlp/extractor/senategov.py]: The bug is fixed

File: yt_dlp/extractor/senategov.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        committee = video_type if video_type == 'arch' else qs['comm'][0]
+
+        stream_num, domain = _COMMITTEES[committee]
+
+        formats = []
+        if video_type == 'arch':
+            filename = video_id if '.' in video_id else video_id + '.mp4'
+            m3u8_url = compat_urlparse.urljoin(domain, 'i/' + filename + '/master.m3u8')
+            for entry in self._extract_m3u8_formats(m3u8_url, video_id, ext='mp4', m3u8_id='m3u8'):
+                formats.append(entry)
```
Comment (pukkandan) [inline yt_dlp/extractor/senategov.py:129]:
```suggestion
            formats = self._extract_m3u8_formats(m3u8_url, video_id, ext='mp4', m3u8_id='m3u8')
```

**Review by pukkandan — COMMENTED** [review]
Are you sure the old URL pattern no longer works for any video?
```py
compat_urlparse.urljoin(domain, filename) + '?v=3.1.0&fp=&r=&g=',
```

**Comment by Grabien** [comment]
We download videos from the senate.gov website often enough, and I also checked some old videos. None of them have the "v=3.1.0" signature.

**Pull request merged**